pkg/template/cluster/cluster.go

```diff
@@ -124,3 +133,60 @@ func newAWSClusterCR(clusterID string, c Config) (*infrastructurev1alpha2.AWSClu
 
 	return awsClusterCR, nil
 }
+
+func newAWSControlPlaneCR(clusterID string, controlPlaneID string, c Config) *infrastructurev1alpha2.AWSControlPlane {
+	return &infrastructurev1alpha2.AWSControlPlane{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AWSControlPlane",
+			APIVersion: "infrastructure.giantswarm.io/v1alpha2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controlPlaneID,
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				label.AWSOperatorVersion: c.ReleaseComponents["aws-operator"],
+				label.Cluster:            clusterID,
+				label.ControlPlane:       controlPlaneID,
+				label.Organization:       c.Owner,
+				label.ReleaseVersion:     c.ReleaseVersion,
+			},
+		},
+		Spec: infrastructurev1alpha2.AWSControlPlaneSpec{
+			AvailabilityZones: []string{c.MasterAZ},
+			InstanceType:      defaultMasterInstanceType,
+		},
+	}
+}
+
+func newG8sControlPlaneCR(obj interface{}, clusterID string, controlPlaneID string, c Config) (*infrastructurev1alpha2.G8sControlPlane, error) {
+	runtimeObj, _ := obj.(runtime.Object)
+
+	infrastructureCRRef, err := reference.GetReference(infrastructurev1alpha2scheme.Scheme, runtimeObj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	cr := &infrastructurev1alpha2.G8sControlPlane{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "G8sControlPlane",
+			APIVersion: "infrastructure.giantswarm.io/v1alpha2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controlPlaneID,
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				label.ClusterOperatorVersion: c.ReleaseComponents["cluster-operator"],
```

Comment:
Just FMI where do the release information come from?

Reply:
Short answer: It was there already and used in the Cluster and AWSCluster resource, too.

Longer answer: it's package `github.com/giantswarm/kubectl-gs/pkg/gsrelease` and seems to take information from the `github.com/giantswarm/releases` repository.